src/main/java/org/sonar/plugins/stash/StashRequestFacade.java

```diff
@@ -183,40 +169,51 @@ void postCommentPerIssue(PullRequestRef pr, String sonarQubeURL,
+        String path = getIssuePath(issue);
+        StashCommentReport comments = commentsByFile.get(path);
+        String commentContent = MarkdownPrinter.printIssueMarkdown(issue, config.getSonarQubeURL());
+        Integer issueLine = issue.line();
+        // FIXME move this somewhere else
+        if (issueLine == null) {
+          issueLine = 0;
+        }
 
         // if comment not already pushed to Stash
-        if ((comments != null) &&
-                (comments.contains(issue.printIssueMarkdown(sonarQubeURL), issue.getPath(), issue.getLine()))) {
-          LOGGER.debug("Comment \"{}\" already pushed on file {} ({})", issue.getKey(),
-                  issue.getPath(), issue.getLine());
+        if (comments != null && comments.contains( commentContent, path, issueLine)) {
+          if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("Comment \"{}\" already pushed on file {} ({})", issue.key(),
+                    path, issueLine);
+          }
           continue;  // Next element in "issue_loop"
         }
 
         // check if issue belongs to the Stash diff view
-        String type = diffReport.getType(issue.getPath(), issue.getLine());
+        String type = diffReport.getType(path, issueLine);
         if (type == null) {
-          LOGGER.info("Comment \"{}\" cannot be pushed to Stash like it does not belong to diff view - {} (line: {})",
-                  issue.getKey(), issue.getPath(), issue.getLine());
+          if (LOGGER.isInfoEnabled()) {
+            LOGGER.info("Comment \"{}\" cannot be pushed to Stash like it does not belong to diff view - {} (line: {})",
+                    issue.key(), path, issueLine);
+          }
           continue;  // Next element in "issue_loop"
         }
 
-        long line = diffReport.getLine(issue.getPath(), issue.getLine());
+        long line = diffReport.getLine(path, issueLine);
 
         StashComment comment = stashClient.postCommentLineOnPullRequest(pr,
-                issue.printIssueMarkdown(sonarQubeURL),
-                issue.getPath(),
-                line,
-                type);
+                commentContent, path, line, type);
 
-        LOGGER.debug("Comment \"{}\" has been created ({}) on file {} ({})", issue.getKey(), type,
-                issue.getPath(), line);
+        if (LOGGER.isDebugEnabled()) {
+          LOGGER.debug("Comment \"{}\" has been created ({}) on file {} ({})", issue.key(), type,
+                  path, line);
+        }
 
         // Create task linked to the comment if configured
-        if (taskSeverities.contains(issue.getSeverity())) {
-          stashClient.postTaskOnComment(issue.getMessage(), comment.getId());
+        if (taskSeverities.contains(issue.severity())) {
+          stashClient.postTaskOnComment(issue.message(), comment.getId());
 
-          LOGGER.debug("Comment \"{}\" has been linked to a Stash task", comment.getId());
+          if (LOGGER.isDebugEnabled()) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS134)